adagrams/game.py

```diff
@@ -1,11 +1,127 @@
+import random
+
 def draw_letters():
-    pass
+
+    LETTER_POOL = {
+        'A': 9, 
+        'B': 2, 
+        'C': 2, 
+        'D': 4, 
+        'E': 12, 
+        'F': 2, 
+        'G': 3, 
+        'H': 2, 
+        'I': 9, 
+        'J': 1, 
+        'K': 1, 
+        'L': 4, 
+        'M': 2, 
+        'N': 6, 
+        'O': 8, 
+        'P': 2, 
+        'Q': 1, 
+        'R': 6, 
+        'S': 4, 
+        'T': 6, 
+        'U': 4, 
+        'V': 2, 
+        'W': 2, 
+        'X': 1, 
+        'Y': 2, 
+        'Z': 1
+    }
+    
+    # initiate draw_letters to record and return drawn random letters
+    # initiate letter_freq to check the frequency of drawn out random letters
+    draw_letters_list = []
+    letter_freq = {}
+    
+    # create random letter picking from the letter poll
+    # only append letters that are no bigger than letter poll frequency
+    while len(draw_letters_list) < 10:
+        cur_random_letter = random.choice(list(LETTER_POOL.keys()))
+        if cur_random_letter in letter_freq.keys():
```

Comment:
Notice that calling keys() on a dictionary returns a list of the keys. When you use the `in` operator, the time complexity is O(n) because Python has to check if cur_random_letter is in the list of keys. Additionally, the `in` operator is nested in a while loop, so the time complexity of this method is O(n^2).